### ISSUE TITLE
too big of a neural net??

### ISSUE THREAD

**Issue by Mcilie** [cazala/synaptic#324]
so i am making an NN to classify mnist digits. The problem is that i keep getting an undefined error when training it, here is the code: 
    var sy = require("synaptic");

    // create the network
    var inputLayer = new sy.Layer(28*28);
    var hiddenLayer = new sy.Layer(10);
    var outputLayer = new sy.Layer(1);

    inputLayer.project(hiddenLayer);
    hiddenLayer.project(outputLayer);

    var myNetwork = new sy.Network({
	    input: inputLayer,
	    hidden: [hiddenLayer],
	    output: outputLayer
    });

    // train the network
    var learningRate = .3;
    for (var i = 0; i < 20; i++)
    {
        console.log(i);

        for(var j = 0; j< MyData.length; j++){
            myNetwork.activate(training_set_inputs[j]);
            myNetwork.propagate(learningRate, training_set_inputs[0][j]);
        }
    }

**Comment by Mcilie** [comment]
The data gathering is cut out of this snippet. Im sure the arrays are all in check. i enabled max memory in node js

**Comment by Mcilie** [comment]
also it stops training at around element 784 of the array. i looked in the csv and it looks fine

**Comment by Mcilie** [comment]
ok so stopping before 784 works fine but all the predictions are Nan

**Comment by Pummelchen** [comment]
If your NN is too big consider this:

- Use Binary Cost function
- Train only 1 dataset ahead and enable backprop
- Normalize correctly with boundaries far away from 0-1, use 0.0001 to  0.9999 
- Use a 8700K for max single thread speed (not sure if V8 supports AVX-512 already ?)

If its still to slow move away from CPU bound NNs to GPU which means Tensorflow with a high end Nvidia card like 1080 and up.

**Comment by Mcilie** [comment]
okay, but Im wondering why its only predicting like .003, like there isnt even a variation in the predictions. So while this definitley may be too big to nodejs, im pretty sure there is something wrong with the way I train it as well. Also, what activation function does the NN use by default?

**Comment by Pummelchen** [comment]
Thats right, in my model I have "only" 27 inputs and the error goes super low like 0.0000001.  Debug your training sets by printing each step and each single value for like 2-3 training steps. Its a mess to read but required to find bugs.
